### PR TITLE
NXP-27429: add function to calculate viewport height

### PR DESCRIPTION
**Pull request by semisse**

### CODE REVIEW

File: elements/search/nuxeo-search-form.js [hunk 1]
```diff
@@ -141,15 +149,21 @@ Polymer({
...
 
       .header {
         height: 53px;
         box-shadow: 0 -1px 0 rgba(0, 0, 0, 0.1) inset;
         @apply --layout-horizontal;
         @apply --layout-center;
+        position: absolute;
+        top: 0;
+        left: 0;
+        right: 0;
```
Comment (Gabez0r) [inline elements/search/nuxeo-search-form.js]: Do we need these? Without them, we wouldn't probably need ⬇️

File: elements/search/nuxeo-search-form.js [hunk 2]
```diff
@@ -141,15 +149,21 @@ Polymer({
       .content {
         @apply --layout-flex;
         @apply --layout-vertical;
-        height: calc(100vh - 61px - var(--nuxeo-app-top));
-        width: 293px;
+        width: 100%;
+        padding-top: 53px;
```
Comment (Gabez0r) [inline elements/search/nuxeo-search-form.js]: This

File: elements/search/nuxeo-search-form.js [hunk 3]
```diff
@@ -225,6 +239,11 @@ Polymer({
       #checkbox.paper-checkbox {
         border: 1px solid;
       }
+
+      #search-container {
+        height: calc(var(--vh, 1vh) * 100);
```
[the remark this answers is not in the document]
Comment (semisse) [reply elements/search/nuxeo-search-form.js]: Doesn't work. We should keep this line and remove the one on the host.

File: elements/search/nuxeo-search-form.js [hunk 4]
```diff
@@ -141,15 +149,20 @@ Polymer({
...
 
       .header {
         height: 53px;
         box-shadow: 0 -1px 0 rgba(0, 0, 0, 0.1) inset;
         @apply --layout-horizontal;
         @apply --layout-center;
+        /* position: absolute;
+        top: 0;
+        left: 0;
+        right: 0; */
```
Comment (Gabez0r) [inline elements/search/nuxeo-search-form.js]: Not needed anymore?

**Comment by semisse** [comment]
@Gabez0r regarding your solution for the white strip, I've used the `--nuxeo-sidebar-width` var instead of the hardcoded value.

**Comment by Gabez0r** [comment]
> @Gabez0r regarding your solution for the white strip, I've used the `--nuxeo-sidebar-width` var instead of the hardcoded value.

Perfect! 👌

**Comment by semisse** [comment]
Pipelines ok: https://qa.nuxeo.org/jenkins/job/web-ui-pipeline-sfialho/163/

File: elements/search/nuxeo-search-form.js [hunk 5]
```diff
@@ -141,15 +149,16 @@ Polymer({
       .content {
         @apply --layout-flex;
         @apply --layout-vertical;
-        height: calc(100vh - 61px - var(--nuxeo-app-top));
-        width: 293px;
+        width: 100%;
+        height: calc(100% - 121px);
```
Comment (Gabez0r) [inline elements/search/nuxeo-search-form.js]: Quick question: for consistency's sake, should we still be relying on `--nuxeo-app-top` as well?

Comment (semisse) [reply elements/search/nuxeo-search-form.js]: Yes we can. But we also need to substract the height of the bottom actions

File: elements/search/nuxeo-search-form.js [hunk 6]
```diff
@@ -141,15 +150,16 @@ Polymer({
       .content {
         @apply --layout-flex;
         @apply --layout-vertical;
-        height: calc(100vh - 61px - var(--nuxeo-app-top));
-        width: 293px;
+        width: 100%;
+        height: calc(100% - 68px - var(--nuxeo-sidebar-width));
```
Comment (nmpcunha) [inline elements/search/nuxeo-search-form.js]: Probably next time we may want to be consistent with properties order. Or, at least, don't change it to be easy to review 😅

Comment (semisse) [reply elements/search/nuxeo-search-form.js]: Agree, no need to revert the order.

File: elements/search/nuxeo-search-form.js [hunk 7]
```diff
@@ -141,15 +150,16 @@ Polymer({
       .content {
         @apply --layout-flex;
         @apply --layout-vertical;
-        height: calc(100vh - 61px - var(--nuxeo-app-top));
-        width: 293px;
+        height: calc(100% - 68px - var(--nuxeo-sidebar-width));
```
Comment (Gabez0r) [inline elements/search/nuxeo-search-form.js]: Why are we subtracting the sidebar **width** from the content height?

**Pull request merged**